### DISCUSSION TITLE
Could you explain the usage of webpack-dev-middleware?

### DISCUSSION THREAD

**Discussion by nyngwang** [webpack/webpack#18112] [Q&A] [answered]
Hi! I'm learning webpack and I just learned that `webpack-dev-server` internally uses `webpack-dev-middleware`. My question is: could `webpack-dev-middleware` be used with any backend server framework other than express.js?

From its [README](https://github.com/webpack/webpack-dev-middleware?tab=readme-ov-file#webpack-dev-middleware) they say:

> An **express-style** development middleware for use with [webpack](https://webpack.js.org/) bundles and allows for serving of the files emitted from webpack. This should be used for development only.

**Comment by alexander-akait** [comment, marked as answer]
>  My question is: could webpack-dev-middleware be used with any backend server framework other than express.js?

Yes, but it must be compatible with `connect` - https://github.com/senchalabs/connect. Feel free to update docs, we have tests for it https://github.com/webpack/webpack-dev-middleware/blob/master/test/middleware.test.js#L26

**Reply by nyngwang** [reply]
> Feel free to update docs, we have tests for it [...]

Sure! it's my pleasure :) Thanks for your clarification and the hard work!
